### DISCUSSION TITLE
Desktop first media queries have wrong order

### DISCUSSION THREAD

**Discussion by HenrijsS** [tailwindlabs/tailwindcss#5002] [Help]
I have change the media breakpoints to max-width: 
```
screens: {
    '2xl': { max: '1536px' },
    // => @media (max-width: 1536px) { ... }
    
    xl: { max: '1280px' },
    // => @media (max-width: 1280px) { ... }
    
    lg: { max: '1024px' },
    // => @media (max-width: 1024px) { ... }
    
    md: { max: '768px' },
    // => @media (max-width: 768px) { ... }
    
    sm: { max: '640px' },
    // => @media (max-width: 640px) { ... }
},
```

And I've inserted the classes like this: `<div class="flex w-3/5 gap-5 2xl:w-3/4 lg:w-4/5">`

But when compiling it the 2xl always comes first: http://snpy.in/tI48ty
Notice that it adds some weird \3 before the class.

**Comment by olets** [comment]
> But when compiling it the 2xl always comes first

The order in the config is respected in the final CSS. 2xl comes first because you have 2xl first.

Here's yours https://play.tailwindcss.com/43pJEAgyMR?file=config and here's ordered sm<md<lg<xl<2xl https://play.tailwindcss.com/bZFGTw3keN?file=config. 

>  `<div class="flex w-3/5 gap-5 2xl:w-3/4 lg:w-4/5">`

If you do reverse the order in the config so that 2xl comes after lg, this html will be equivalent to

```
<div class="flex w-3/5 gap-5 2xl:w-3/4">
```

The "not wider than 1536" styles will apply from 1024 to 1536 _and also_ from 1024 down. 

> Notice that it adds some weird \3 before the class.

Class names can't start with a number. `\3` is an escape character. Writing it this way is a workaround to allow class names starting with `2xl`.
